airflow-core/src/airflow/models/dagbag.py

```diff
@@ -143,6 +146,20 @@ def handle_timeout(signum, frame):
             signal.setitimer(signal.ITIMER_REAL, 0)
 
 
+def _validate_executor_fields(dag: DAG) -> None:
+    for task in dag.tasks:
+        if not task.executor:
+            continue
+        try:
+            ExecutorLoader.lookup_executor_name_by_str(task.executor)
+        except UnknownExecutorException:
+            raise UnknownExecutorException(
+                f"The specified executor {task.executor} for task {task.task_id} is not "
+                "configured. Review the core.executors Airflow configuration to add it or "
+                "update the executor configuration for this task."
+            )
```

Comment:
```suggestion
            raise UnknownExecutorException(
                f"Task '{task.task_id}' specifies executor '{task.executor}', which is not available. "
                "Make sure it is listed in your [core] executors configuration, or update the task's "
                "executor to one of the configured executors."
            )
```

Slightly clearer, nit